src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/_validators.py

```diff
@@ -75,8 +76,10 @@ def get_config_value(section, key, default):
     # if connection string supplied or in environment variables, extract account key and name
     if n.connection_string:
         conn_dict = validate_key_value_pairs(n.connection_string)
-        n.account_name = conn_dict['AccountName']
-        n.account_key = conn_dict['AccountKey']
+        n.account_name = conn_dict.get('AccountName')
+        n.account_key = conn_dict.get('AccountKey')
+        if not n.account_name or not n.account_key:
+            raise CLIError('Connection-string: %s, is malformed.' % n.connection_string)
```

Comment:
You could make this more information (but you don't have to):

`Connection-string: '<CS>' is malformed. Missing: <props>. Double-check the connection string. Some shell environments require the connection string to be surrounded by quotes.`